hedera-node/src/main/java/com/hedera/services/txns/consensus/TopicCreateTransitionLogic.java

```diff
@@ -158,32 +120,20 @@ private ResponseCodeEnum validatePreSignatureValidation(TransactionBody transact
 
 	/**
 	 * Validation of the post-consensus transaction just prior to state transition.
-	 * @return the validity
+	 * Throws {@link com.hedera.services.exceptions.InvalidTransactionException} on failed validation.
 	 */
-	private ResponseCodeEnum validatePreStateTransition() {
-		var op = transactionContext.accessor().getTxn().getConsensusCreateTopic();
-
-		ResponseCodeEnum validationResult = validator.memoCheck(op.getMemo());
-
-		if (op.hasSubmitKey() && !validator.hasGoodEncoding(op.getSubmitKey())) {
-			validationResult = BAD_ENCODING;
-		} else if (!op.hasAutoRenewPeriod()) {
-			validationResult = INVALID_RENEWAL_PERIOD;
-		} else if (!validator.isValidAutoRenewPeriod(op.getAutoRenewPeriod())) {
-			validationResult = AUTORENEW_DURATION_NOT_IN_RANGE;
-		} else if (op.hasAutoRenewAccount()) {
-			final var reqAutoRenew = op.getAutoRenewAccount();
-			final var sanityCheck = validator.queryableAccountStatus(reqAutoRenew, accounts.get());
-			if (sanityCheck != OK) {
-				return INVALID_AUTORENEW_ACCOUNT;
-			}
-			if (ledger.isDetached(reqAutoRenew)) {
-				validationResult = ACCOUNT_EXPIRED_AND_PENDING_REMOVAL;
-			} else if (!op.hasAdminKey()) {
-				validationResult = AUTORENEW_ACCOUNT_NOT_ALLOWED;
-			}
+	private void validatePreStateTransition() {
```

Comment:
Why don't we use the precheck validation as we do in the rest of the transition logics?

Reply:
Will do :)

Reply:
I think there is a bit of misunderstanding here.
The `validatePreStateTransition` is actually a `post-consensus` validation, executed right before the state transition. 

The `validate` method is actually the `precheck ` validation. We will drastically change the API if we perform post-consensus validation in the precheck ( exactly 7 failing e2e specs ) .

**TL;DR:**

`TransitionLogic.validateSemantics` is the precheck, in which the API is not expecting us to validate accounts. 
`validatePreStateTransition` **is** the transition logic validation, like `validateMinting` in the `TokenMintTransitionLogic`